src/ansys/geometry/core/tools/repair_tools.py

```diff
@@ -254,8 +289,106 @@ def find_stitch_faces(self, bodies: List["Body"]) -> List[StitchFaceProblemAreas
         problem_areas_response = self._repair_stub.FindStitchFaces(
             FindStitchFacesRequest(faces=body_ids)
         )
-        problem_areas = [
-            StitchFaceProblemAreas(res.id, list(res.body_monikers), self._grpc_client)
+        parent_design = RepairTools.get_root_component(bodies[0])
+        return [
+            StitchFaceProblemAreas(
+                res.id,
+                self.get_bodies_from_ids(parent_design, res.body_monikers),
+                self._grpc_client,
+            )
             for res in problem_areas_response.result
         ]
-        return problem_areas
+
+    def get_bodies_from_ids(self, design: "Design", body_ids: List[str]):
+        """
+        Find the body object from its id.
+
+        This method takes a design and body ids and gets their corresponding face object.
+
+        Parameters
+        ----------
+        design : Design
+            Parent design for the faces.
+
+        body_ids : List[str]
+            List of body ids corresponding to the problem area.
+
+        Returns
+        -------
+        List[Body]
+            List of face objects representing duplicate face problem areas.
+        """
+        return [body for body in design.bodies if str(body.id) in set(map(str, body_ids))]
+
+    def get_faces_from_ids(self, design: "Design", face_ids: List[str]):
+        """
+        Find the face object from its id.
+
+        This method takes a design and face ids and gets their corresponding face object.
+
+        Parameters
+        ----------
+        design : Design
+            Parent design for the faces.
+
+        face_ids : List[str]
+            List of face ids corresponding to the problem area.
+
+        Returns
+        -------
+        List[Face]
+            List of face objects representing duplicate face problem areas.
+        """
+        return [
+            face
+            for body in design.bodies
+            for face in body.faces
+            if str(face.id) in set(map(str, face_ids))
+        ]
+
+    def get_edges_from_ids(self, design: "Design", edge_ids: List[str]):
+        """
+        Find the edge object from its id.
+
+        This method takes a design and edge ids and gets their corresponding face object.
+
+        Parameters
+        ----------
+        design : Design
+            Parent design for the faces.
+
+        edge_ids : List[str]
+            List of edge ids corresponding to the problem area.
+
+        Returns
+        -------
+        List[Edge]
+            List of edge objects representing duplicate face problem areas.
+        """
+        return [
+            edge
+            for body in design.bodies
+            for face in body.faces
+            for edge in face.edges
+            if str(edge.id) in set(map(str, edge_ids))
```

Comment:
Why are you using set(map(...)) instead of using the list of ``edge_ids`` directly to verify whether the id is inside or not? Also, why are you doing str(edge.id)? It is already a string...